test/SparqlExpressionTest.cpp

```diff
@@ -1242,6 +1242,17 @@ TEST(SparqlExpression, ReplaceExpression) {
       idOrLitOrStringVec({"null", "Xs", "zwei", "drei", U, U}),
       std::tuple{idOrLitOrStringVec({"null", "eins", "zwei", "drei", U, U}),
                  IdOrLiteralOrIri{lit("e.[a-z]")}, IdOrLiteralOrIri{lit("X")}});
+  // A regex with replacement with substitutions
+  checkReplace(
+      idOrLitOrStringVec({"\"$1 \\\\2 A \\\\bc\"", "\"$1 \\\\2 DE \\\\f\""}),
+      std::tuple{idOrLitOrStringVec({"Abc", "DEf"}),
+                 IdOrLiteralOrIri{lit("([A-Z]+)")},
+                 IdOrLiteralOrIri{lit("\"\\\\$1 \\\\2 $1 \\\\\"")}});
```

Comment:
These should definitely be refactored to uuse raw string literals, I am too stupid to read this.

Reply:
Raw string literals will only help a little bit. The problem is that the iri function expects a string in SPARQL iri syntax. So if I want to pass a `\`, I have to pass it as `\\`, and because we write the code in C++ we have to write `\\\\`. So yeah, the raw string literal will halve the amount of backslahes, but that's still too many of them.